gidgethub/actions.py

```diff
@@ -43,3 +42,30 @@ def command(cmd: str, data: str = "", **parameters: str) -> None:
     )
     cmd_parts.append(f"::{data}")
     print("".join(cmd_parts))
+
+
+_DELIMITER = "__GIDGETHUB_DELIMITER__"
+
+
+def setenv(name: str, value: str) -> None:
+    """Create or update an environment variable.
+
+    The change applies to this action and future actions running in the job.
+    """
+    # https://github.com/actions/toolkit/blob/af821474235d3c5e1f49cee7c6cf636abb0874c4/packages/core/src/core.ts#L35-L53
+    os.environ[name] = value
+    # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#multiline-strings
+    write_value = f"{name}<<{_DELIMITER}{os.linesep}{value}{os.linesep}{_DELIMITER}"
+    with open(os.environ["GITHUB_ENV"], "a", encoding="utf-8") as file:
+        file.write(write_value + os.linesep)
+
+
+def addpath(path: Union[str, pathlib.Path]) -> None:
+    """Prepend to PATH.
+
+    This affects this action and all subsequent actions in the current job.
+    """
+    # https://github.com/actions/toolkit/blob/af821474235d3c5e1f49cee7c6cf636abb0874c4/packages/core/src/core.ts#L63-L75
+    os.environ["PATH"] = f"{path!s}{os.pathsep}{os.environ['PATH']}"
+    with open(os.environ["GITHUB_PATH"], "a", encoding="utf-8") as file:
+        file.write(str(path) + os.linesep)
```

Comment:
Calling `str()` when expecting a file path is dangerous as someone who isn't type checking may pass in `None` which will still succeed.

```suggestion
        file.write(os.fspath(path) + os.linesep)
```